### PR TITLE
Add a writer for DNS settings

### PR DESCRIPTION
**Pull request by imobachgs**

### CODE REVIEW

File: test/y2network/config_writer/sysconfig_dns_test.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+describe Y2Network::ConfigWriter::SysconfigDNS do
+  subject(:writer) { described_class.new }
+
+  describe "#write" do
+
+    let(:dhcp_hostname) { true }
+    let(:dns) do
+      Y2Network::DNS.new(
+        name_servers:       [IPAddr.new("10.0.0.1"), IPAddr.new("10.0.0.2")],
```
Comment (teclator) [inline test/y2network/config_writer/sysconfig_dns_test.rb]:
```suggestion
        nameservers:       [IPAddr.new("10.0.0.1"), IPAddr.new("10.0.0.2")],
```

File: test/y2network/config_writer/sysconfig_dns_test.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        search_domains:     ["example.net", "example.org"],
+        resolv_conf_policy: "auto",
+        dhcp_hostname:      dhcp_hostname
+      )
+    end
+
+    let(:old_dhcp_hostname) { !dhcp_hostname }
+    let(:old_dns) do
+      Y2Network::DNS.new(
+        name_servers:       [IPAddr.new("10.0.0.1")],
```
Comment (teclator) [inline test/y2network/config_writer/sysconfig_dns_test.rb]:
```suggestion
        nameservers:       [IPAddr.new("10.0.0.1")],
```

File: src/lib/y2network/config_writer/sysconfig_dns.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_POLICY"),
+          dns.resolv_conf_policy
+        )
+        Yast::SCR.Write(
+          Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SEARCHLIST"),
+          dns.search_domains.join(" ")
+        )
+        Yast::SCR.Write(
+          Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SERVERS"),
+          dns.name_servers.join(" ")
```
Comment (teclator) [inline src/lib/y2network/config_writer/sysconfig_dns.rb]:
```suggestion
          dns.nameservers.join(" ")
```

File: test/y2network/config_writer/sysconfig_dns_test.rb [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    it "updates the list of search domains" do
+      expect(Yast::SCR).to receive(:Write)
+        .with(
+          Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SEARCHLIST"),
+          "example.net example.org"
+        )
+      writer.write(dns, old_dns)
+    end
+
+    it "updates the list of name servers" do
```
Comment (teclator) [inline test/y2network/config_writer/sysconfig_dns_test.rb]:
```suggestion
    it "updates the list of nameservers" do
```

Comment (imobachgs) [reply test/y2network/config_writer/sysconfig_dns_test.rb]: It is true that the variable is called `nameserver`, but we use `name server` in the UI, so I will keep it here as it is (against you have a strong opinion against).

**Comment by teclator** [comment]
LGTM

**Pull request merged**